imagedephi/utils/cli.py

```diff
@@ -15,3 +17,32 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
         return asyncio.run(f(*args, **kwargs))
 
     return wrapper
+
+
+class FallthroughGroup(click.Group):
+    """A Group which may run a subcommand when no subcommand is specified."""
+
+    def __init__(self, subcommand_name: str, should_fallthrough: Callable[[], bool], **attrs: Any):
+        # Subcommands are not added until after this is instantiated,
+        # so only store the future subcommand name
+        self.subcommand_name = subcommand_name
+        self.should_fallthrough = should_fallthrough
+
+        attrs["invoke_without_command"] = True
+        attrs["no_args_is_help"] = False
```

Comment:
Just making sure I understand - this subclass of `click.Group` behaves the same as the stock `click.Group`, except if its being run from the windows explorer, it interprets no arguments as running "fallthrough" command instead of just printing the help text?

Reply:
Yes, but the class is more generic, which makes it easier to test and reason about. If there are no arguments and the value of the `should_fallthrough` is true, then it runs the specified subcommand instead of printing the help text.